worker/worker.py

```diff
@@ -461,6 +413,45 @@ def my_error(e):
         return None
 
     # Step 5: fixup the config options.
+    e = expression.Expression_Parser(
+        variables={"max": max_cpu_count}, functions={"min": min}
+    )
+    try:
+        options.concurrency = round(e.parse(options.concurrency_))
+    except:
+        print("Unable to parse expression for concurrency")
+        return None
+
+    if options.concurrency <= 0:
+        print("--concurrency must be at least 1")
+        return None
+
+    if (
+        "max" not in options.concurrency_ and options.concurrency >= max_cpu_count
+    ) or options.concurrency > max_cpu_count:
+        print(
+            (
+                "\nYou cannot have concurrency {} but at most:\n"
+                "  a) {} with '--concurrency max';\n"
+                "  b) {} otherwise.\n"
+                "Please use option a) only if your computer is very lightly loaded.\n"
+            ).format(
+                options.concurrency,
+                max_cpu_count,
+                max_cpu_count - 1,
+            )
+        )
+        return None
+
+    e = expression.Expression_Parser(
+        variables={"max": mem / 1024 / 1024}, functions={"min": min}
+    )
+    try:
+        options.max_memory = round(e.parse(options.max_memory_))
```

Comment:
Just to avoid to report a worker with TB of ram :)
```python
        options.max_memory = round(
            max(min(e.parse(options.max_memory_), mem / 1024 / 1024), 0)
        )
```

Reply:
Changed!